src/processors/processCherishedTransfer.ts

```diff
@@ -0,0 +1,126 @@
+import { TechRecordType } from '@dvsa/cvs-type-definitions/types/v3/tech-record/tech-record-verb';
+import { setCreatedAuditDetails, setLastUpdatedAuditDetails } from '../services/audit';
+import { UserDetails } from '../services/user';
+import { StatusCode } from '../util/enum';
+
+export const processCherishedTransfer = (
+  userDetails: UserDetails,
+  newVrm: string,
+  recipientRecord: TechRecordType<'get'>,
+  newDonorVrm?: string,
+  donorRecord?: TechRecordType<'get'>,
+)
+: TechRecordType<'get'>[] => {
+  if (newDonorVrm && donorRecord) {
+    return cherishedTransferWithDonor(
+      userDetails,
+      newVrm,
+      newDonorVrm,
+      recipientRecord,
+      donorRecord,
+    );
+  }
+  return cherishedTransferNoDonorVehicle(newVrm, recipientRecord, userDetails);
+};
+
+const formatSecondaryVrms = (record:TechRecordType<'get'>): Array<string> | undefined => {
+  if (record.techRecord_vehicleType !== 'trl') {
+    const secondaryVrms: string[] = record.secondaryVrms ? [...record.secondaryVrms] : [];
+    secondaryVrms.push(record.primaryVrm ?? '');
+    return secondaryVrms;
+  }
+  return undefined;
+};
+
+const cherishedTransferNoDonorVehicle = (
+  newVrm: string,
+  recipientRecord: TechRecordType<'get'>,
+  userDetails: UserDetails,
+): TechRecordType<'get'>[] => {
+  const recordToArchive = { ...recipientRecord };
+  const newRecord = { ...recipientRecord };
+  const recipientOldVrms = formatSecondaryVrms(newRecord);
+
+  const updatedRecordToArchive = setLastUpdatedAuditDetails(
+    recordToArchive,
+    userDetails.username,
+    userDetails.msOid,
+    new Date().toISOString(),
+    StatusCode.ARCHIVED,
+  );
+
+  const updatedNewRecord = setCreatedAuditDetails(
+    newRecord,
+    userDetails.username,
+    userDetails.msOid,
+    new Date().toISOString(),
+    recipientRecord.techRecord_statusCode as StatusCode,
+  );
+  if (updatedNewRecord.techRecord_vehicleType !== 'trl') {
+    updatedNewRecord.primaryVrm = newVrm.toUpperCase();
+    updatedNewRecord.secondaryVrms = recipientOldVrms;
+  }
+  updatedNewRecord.techRecord_reasonForCreation = 'Update VRM - Cherished Transfer';
+
+  return [updatedNewRecord, updatedRecordToArchive, {} as TechRecordType<'get'>, {} as TechRecordType<'get'>];
```

Comment:
I would definitely avoid using this casting of an empty object to `TechRecordType<"get">` here. This means that we're passing an empty object down to the database service. I would think `dynamoDB` is not too happy if you tell it to `put` two empty objects.

The first thing that comes to mind it to create an interface such as
```ts
interface CherishedTransferRequest {
  recordsToUpdate: TechRecordType<"get">[],
  recordsToArchive: TechRecordType<"get">[],
} 
```
And then you can put the records in their relevant spots, regardless of whether there's a donor vehicle of not, and just pass that to the `updateVehicle` function. TBH you probably don't even need the interface, just an object would work.

Reply:
this would also probably allow you to bypass having two different functions for with donor or without donor, but instantiating the object first and then pushing to those arrays of `update` or `archive` ad hoc. 

```ts
const request = {recordsToUpdate: [], recordToArchive: []}

if (donorVehicle) {
// do some logic here
request.recordsToUpdate.push(updatedDonorRecord)
request.recordsToArchived.push(archivedDonorRecord)
}

// some more logic
request.recordsToUpdate.push(updatedNewVehicle)
request.recordsToArchived.push(archivedNewVehicle)

return request
```